model/lifecycle.go

```diff
@@ -930,11 +931,42 @@ func createTasksForBuild(creationInfo TaskCreationInfo) (task.Tasks, error) {
 	return tasks, nil
 }
 
+// addSingleHostTaskGroupDependencies adds dependencies to tasks in a single-host task group
+func addSingleHostTaskGroupDependencies(taskMap map[string]*task.Task, p *Project, taskIds TaskIdTable) {
+	for _, t := range taskMap {
+		if t.TaskGroup == "" {
+			continue
+		}
+		tg := p.FindTaskGroup(t.TaskGroup)
+		if tg == nil || tg.MaxHosts > 1 {
+			continue
+		}
+		singleHostTGDeps := []TaskUnitDependency{}
+		// Iterate backwards until we find a task that exists in the taskMap. This task
+		// will be the parent dependency for the current single host TG task.
+		taskFound := false
+		for i := len(tg.Tasks) - 1; i >= 0; i-- {
+			if t.DisplayName == tg.Tasks[i] {
```

Comment:
Not sure how much this matters really, but would this account for tasks that are used multiple times in the same task group? For example:
```yaml
task_groups:
    - name: my_task_group
      tasks:
        - run-my-script
        - test-a-thing
        - run-my-script
```
I don't know if this is necessarily a case we need to handle or perhaps document.

Reply:
I don't think we need to handle this since it is a [validation error](https://gist.github.com/hadjri/73759d0ef290638c1b637b14b7e33699#file-project_validator-go-L1504) to do that. I'll add a comment to make it clear we can rely on this fact though